components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/service/DeviceOrganizationProviderServiceImpl.java

```diff
@@ -0,0 +1,354 @@
+            if (tempHolder.getDeviceId().equals(SERVER_ID)) {
+                serverFlag++;
+            }
+        }
+        if (serverFlag == 0) {
+            try {
+                DeviceManagementDAOFactory.beginTransaction();
+                deviceOrganizationDAOimpl.addDeviceOrganization(SERVER_ID, "WSO2 IoT server",
+                        "", 0, 2, 1);
+                DeviceManagementDAOFactory.commitTransaction();
+            } catch (TransactionManagementException e) {
+                String msg = "Error occurred while initiating transaction";
+                log.error(msg, e);
+            } catch (DeviceOrganizationDAOException e) {
+                DeviceManagementDAOFactory.rollbackTransaction();
+                String msg = "Error occurred while adding server Node to visualization";
+                log.error(msg, e);
+            } finally {
+                DeviceManagementDAOFactory.closeConnection();
+            }
+        }
+        List<DeviceOrganizationMetadataHolder> newTempDevicesInOrganization = new ArrayList<>();
+        try {
+            newTempDevicesInOrganization = this.getDevicesInOrganization();
+        } catch (DeviceOrganizationException e) {
+            String msg = "Error while getting devices in organization database";
+            log.error(msg, e);
+        }
+        for (DeviceOrganizationMetadataHolder tempHolder : newTempDevicesInOrganization) {
+            String tempId = tempHolder.getDeviceId();
+            String tempLabel = tempHolder.getDeviceName();
+            int tempSize = 21;
+            String tempColor = "#0000ff";
+            if (tempHolder.getDeviceId().equals(SERVER_ID)) {
+                tempColor = "#ffa500";
+                tempSize = 70;
+            }
+            if (!tempHolder.getDeviceId().equals(SERVER_ID) && tempHolder.getIsGateway() == 1) {
+                tempSize = 33;
+                tempColor = "#ff0000";
+            }
+            if (tempHolder.getState() == 1 || tempHolder.getState() == 0) {
+                tempColor = "#d3d3d3";
+            }
+            nodes.add(new DeviceOrganizationVisNode(tempId, tempLabel, tempSize, tempColor));
+        }
+        return nodes;
+    }
+
+    @Override
+    public List<DeviceOrganizationVisEdge> generateEdges() {
+        List<DeviceOrganizationMetadataHolder> tempDevicesInOrganization = new ArrayList<>();
+        try {
+            tempDevicesInOrganization = this.getDevicesInOrganization();
+        } catch (DeviceOrganizationException e) {
+            String msg = "Error while getting devices in organization";
+            log.error(msg, e);
+        }
+        List<DeviceOrganizationVisEdge> edges = new ArrayList<>();
+        for (DeviceOrganizationMetadataHolder tempHolder : tempDevicesInOrganization) {
+            String child = tempHolder.getDeviceId();
+            String parent = tempHolder.getParent();
+            if (parent.equals("") || parent.equals(" ")) {
```

Comment:
I think it is better if you could use Yoda Expression here. That is  " ".equals(tempHolder.getParent());  So that it is no need to check if tempHolder.getParent() is null.

Reply:
Used another technique suggested by @Kamidu first trimming then checking for empty values. Refer 9d4b1663c67e34add2e5e6ce2c83150768f4fee9